src/api-binder/report.ts

```diff
@@ -138,13 +152,25 @@ export async function startReporting() {
 		'appUpdatePollInterval',
 	])) as StateReportOpts;
 
-	let reportPending = false;
+	const throttledReport = throttle(
+		reportCurrentState,
+		constants.maxReportFrequency,
+	);
+
 	const doReport = async () => {
-		if (!reportPending) {
-			reportPending = true;
-			await reportCurrentState(reportConfigs);
-			reportPending = false;
-		}
+		return new Promise<void>((resolve) => {
+			throttledReport(reportConfigs, (result: ReportResult) => {
+				if (result.reported) {
+					log.info('Reported current state to the cloud');
+					stateReportErrors = 0;
+					lastReport = result.state.full;
+				} else {
+					// No report was sent to reset the throttle
+					throttledReport.cancel();
+				}
+				resolve();
+			});
+		});
```

Comment:
This returns a promise so that the callback and resolve the function in order to retain the recursivelyReport function below. That function will report and wait until it completes before delaying until the next report.